packages/mui-joy/src/utils/GenerateCssVarsType.ts

```diff
@@ -0,0 +1,12 @@
+/**
+ * A type which generates CSS Variables.
+ * @example
+ * type ButtonCssVars = GenerateCssVarsType<{color:"--Button-color";size:"--Button-size"}>;
+ *  ButtonCssVars would be {
+ * '--Button-color'?: string | number;
+ * '--Button-size'?: string | number;
+ * }
+ */
+export type GenerateCssVarsType<CssVars extends Record<string, string>> = {
+  [Key in keyof CssVars as CssVars[Key]]?: string | number;
+};
```

Comment:
This is what I was referring to in question one. 

The same code is in this [pull](https://github.com/mui/material-ui/pull/39044), from which I am continuing on.

Reply:
Yes, once that pr is merged is you can remove this commit.

Right now, team is discussing approaches to provide type and approach is not finalized yet. Once it is done and merged you can work on other components

Reply:
Awesome! Thanks for the timely response, I'll keep an eye on the issue then.

If you don't mind me asking, can you take a quick second to help me understand the pull check errors? 
- labels, this I presume you guys have to do as I don't have access to label options.
- Argos, this is a little confusing as the components I modified aren't the ones that failed the check (at least i think so)
- ci/circleci, I presume this is because there is no spec.ts for linear progression, is this something I should add.


_I get that this might all be a waste of time if the team decides to go another direction, but I've enjoyed the process so far so all is not lost_